crates/turborepo-auth/src/login_server.rs

```diff
@@ -20,39 +31,71 @@ pub struct SsoPayload {
 }
 
 #[async_trait]
-pub trait SSOLoginServer {
-    async fn run(&self, port: u16, verification_token: Arc<OnceCell<String>>) -> Result<(), Error>;
+pub trait LoginServer {
```

Comment:
I'm a little confused. What's the rationale behind moving to a single function with a `login_type` parameter?

Reply:
Mostly around the `LoginOptions`. I wasn't sure of how to make it so that the server field was either an SSO server or basic Login server, so I just collapsed them into a single LoginServer trait. Open to cleaner options here